aten/src/ATen/native/native_functions.yaml

```diff
@@ -2290,26 +2290,13 @@
   device_check: NoCheck
   device_guard: False
 
-# decomposes to eye.m
-- func: eye(int n, *, ScalarType? dtype=None, Layout? layout=None, Device? device=None, bool? pin_memory=None) -> Tensor
+- func: eye.m(int n, int? m=None, *, int k=0, ScalarType? dtype=None, Layout? layout=None, Device? device=None, bool? pin_memory=None) -> Tensor
   dispatch:
     CompositeExplicitAutograd: eye
 
-- func: eye.m(int n, int m, *, ScalarType? dtype=None, Layout? layout=None, Device? device=None, bool? pin_memory=None) -> Tensor
+- func: eye.m_out(int n, int? m=None, *, int k=0, Tensor(a!) out) -> Tensor(a!)
```

Comment:
Unfortunately I think this will be a BC issue, and we should probably maintain the current ATen signatures + add a note. The internal implementation of the signatures can call a new `eye.m_out` op, if desired.

Reply:
Ok, I'll revert those changes and mark them with a sad face in `native_functions.yaml`.

Reply:
After a bit of thinking (and after having added the relevant tests to the OpInfos) I don't think there's a way to implement this function so that it works in both Python an C++, and is also BC. See this comment from Ivan for an example where the previous implementation breaks https://github.com/pytorch/pytorch/pull/72443/files#r924345868.

Reply:
cc @IvanYashchuk in case he has any ideas.

Reply:
`LazyTensor` complains as well at https://github.com/pytorch/pytorch/actions/runs/3090061256/jobs/4998701127

Reply:
Nevermind, I fixed the `LazyTensor` issue. Now, I don't know how to go about the JIT, or whether it can be fixed at all...